moveit_simple/src/moveit_simple.cpp

```diff
@@ -611,6 +619,50 @@ void Robot::clearTrajectory(const::std::string traj_name)
 
 
 
+moveit_simple::JointTrajectoryType Robot::plan(const std::string traj_name, 
+                                               bool collision_check)
+{
+  std::lock_guard<std::recursive_mutex> guard(m_);
+
+  if ( traj_info_map_.count(traj_name) )
+  {
+    JointTrajectoryType goal;
+    goal.trajectory.joint_names = joint_group_->getVariableNames();
+    try
+    {
+      if ( toJointTrajectory(traj_name, goal.trajectory.points, collision_check) )
+      {
+
+        // Modify the speed of execution for the trajectory based off of the speed_modifier_
+        for (std::size_t i = 0; i < goal.trajectory.points.size(); i++)
+        {
+          goal.trajectory.points[i].time_from_start *= (1.0/speed_modifier_);
```

Comment:
The trajectory scaling by the `speed_modifier_` should either occur in the `plan` method or the `execute` method, but not both.  I'd recommend the `plan` method because that was the whole reason for pulling the `speed_modifier_` into the `Robot` base type.